### PR TITLE
Inventory fix

### PR DESCRIPTION
**Pull request by cre2016**
Fix can't pickup item for more than 36 inventory .
Fix player can pickup item when his inventory is full .
Fix didn't remove item on right index in his inventory , when player drop item .
Fix change hotbar index then one of the hotbarslot didn't change ,when the two item both on hotbar .

### CODE REVIEW

File: src/pocketmine/Player.php [hunk 1]
```diff
@@ -3293,7 +3293,7 @@ public function handleDataPacket(DataPacket $packet){
 					break;
 				}
 
-				$dropItem = $this->inventory->contains($packet->item) ? $packet->item : $this->inventory->getItemInHand();
+				$dropItem = $item = $this->inventory->getItemInHand();
```
Comment (ishitatsuyuki) [inline src/pocketmine/Player.php]: WRONG.

File: src/pocketmine/Player.php [hunk 2]
```diff
@@ -3302,6 +3302,7 @@ public function handleDataPacket(DataPacket $packet){
 				}
 
 				$this->inventory->remove($dropItem);
+				$this->inventory->setItemInHand(Item::get(Item::AIR, 0, 1));
```
Comment (ishitatsuyuki) [inline src/pocketmine/Player.php]: Why you need this?

File: src/pocketmine/inventory/PlayerInventory.php [hunk 3]
```diff
@@ -158,7 +158,8 @@ public function onSlotChange($index, $before){
 		if($index >= $this->getSize()){
 			$this->sendArmorSlot($index, $this->getViewers());
 			$this->sendArmorSlot($index, $this->getHolder()->getViewers());
-		}elseif($holder instanceof Player){
+		};
```
Comment (ishitatsuyuki) [inline src/pocketmine/inventory/PlayerInventory.php]: Looks unneed.

File: src/pocketmine/Player.php (hunk 2 above)
Comment (cre2016) [inline src/pocketmine/Player.php]: set the item in hand

File: src/pocketmine/inventory/PlayerInventory.php (hunk 3 above)
Comment (cre2016) [inline src/pocketmine/inventory/PlayerInventory.php]: "elseif"  is only work when " if " isn't  work

Comment (ishitatsuyuki) [inline src/pocketmine/inventory/PlayerInventory.php]: You only need to send one of armor or inventory.

File: src/pocketmine/Player.php (hunk 2 above)
Comment (ishitatsuyuki) [inline src/pocketmine/Player.php]: The item is already removed. This is possibly doing nothing and may cause further bugs.

**Comment by ishitatsuyuki** [comment]
Fix the points I pointed out and do a rebase.

File: src/pocketmine/Player.php (hunk 2 above)
Comment (cre2016) [inline src/pocketmine/Player.php]: That's important ,can't be deleted ! It will cause BUG .I have tested it .

**Comment by cre2016** [comment]
where to run these rebase commands ?

**Comment by ishitatsuyuki** [comment]
What? You even cannot type commands?

Run in your working directory.

**Comment by cre2016** [comment]
I only fork and commit on github .And I had test it in my computer .

**Comment by ishitatsuyuki** [comment]
Will merge manually. Closing.

**Pull request closed without merging**